guardian/manage-frontend
# Remove redundant usage of  mdapiResponseReader

## Background
Work was done to add user details to the MMA response. For the work to be shipped successfully we had to enable the client side to accept the previous response (a JSON array of products) and the new response which contains the user

This ticket is to tidy up the code base to remove the old response type and do any refactoring which may be required.

## What does this change?
Removed mdapiResponseReader from productResponse which then cascaded into a refactor to remove where it was called and only pass in the new MembersDataApiResponse object. 

Some of the tests needed to be refactored to used the Test Utils object which is the new response object and irrelevant tests have been removed since it tested the mdapiResponseReader.

## How to test
On `local` or `CODE` make sure all of the screens and process are working as expected. We hope that there is no change in behaviour or errors in the logs.
